### PR TITLE
nrf_cloud: unittest: editted cloud unit tests to be run automatically

### PR DESCRIPTION
**Pull request by tony-le-24**
Added integration platforms of qemu_cortex_m3 and native_posix, also added Kconfigs overlay for qemu_cortex_m3. These will make the unit tests run automatically with sdk-nrf integration.

See IRIS-4986

Signed-off-by: Tony Le <tony.le@nordicsemi.no>

### CODE REVIEW

**Comment by tony-le-24** [comment]
need inquiries from @balaji-nordic to fix twister check.

File: tests/subsys/net/lib/nrf_cloud/cloud/prj_qemu_cortex_m3.conf [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+# Disable sockets and POSIX wrappers
+CONFIG_NET_SOCKETS=n
+CONFIG_NET_SOCKETS_POSIX_NAMES=n
+
+# Dependencies
+CONFIG_NEWLIB_LIBC=y
+CONFIG_NEWLIB_LIBC_FLOAT_PRINTF=y
+
+# qemu_cortex_m3
```
Comment (balaji-nordic) [inline tests/subsys/net/lib/nrf_cloud/cloud/prj_qemu_cortex_m3.conf:22]: Have a more descriptive comment than this. May be add a reference to https://github.com/zephyrproject-rtos/zephyr/issues/15565 ?

**Review by balaji-nordic — CHANGES_REQUESTED** [review]

**Review by jayteemo — APPROVED** [review]

File: tests/subsys/net/lib/nrf_cloud/cloud/prj_qemu_cortex_m3.conf (hunk 1 above)
Comment (tony-le-24) [reply tests/subsys/net/lib/nrf_cloud/cloud/prj_qemu_cortex_m3.conf:22]: changed to "# For qemu_cortex_m3 to run successfully, it is a fix in zephyrproject-rtos/zephyr#15565"

**Review by plskeggs — APPROVED** [review]

**Review by balaji-nordic — APPROVED** [review]
I suggested a small update to the wording. Looks good otherwise.

**Pull request merged**